lib/Wikia/src/Service/User/Permissions/PermissionsServiceImpl.php

```diff
@@ -0,0 +1,391 @@
+<?php
+
+namespace Wikia\Service\User\Permissions;
+
+class PermissionsServiceImpl implements PermissionsService {
+
+	/** @var  PermissionsConfiguration */
+	private $permissionsConfiguration;
+
+	/** @var string[string] - key is user id */
+	private $localExplicitUserGroups = [];
+
+	/** @var string[string] - key is the user id */
+	private $globalExplicitUserGroups = [];
+
+	/** @var string[string] - key is the user id */
+	private $implicitUserGroups = [];
+
+	/** @var string[string] - key is user id */
+	private $userPermissions = [];
+
+	public function __construct() {
+		$this->permissionsConfiguration = new PermissionsConfigurationImpl();
+	}
+
+	public function getConfiguration() {
+		return $this->permissionsConfiguration;
+	}
+
+	public function getExplicitGroups(\User $user ) {
+		return array_unique( array_merge (
```

Comment:
I could store it in internal variable, similarly to the groups read from the DB but then I need to remember about purging it when the underlying data changes and I'm not convinced it's worth it. I would totally agree if this was a DB operation etc. but here I'm not convinced it's worth the extra code complexity, especially since in practice these arrays rarely have more then a few elements.